# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/cookiebanners/components.conf
# Commit: 7976b0cb00d6
# Full Hash: 7976b0cb00d60fe63424c3572e0c64a6ec0bebcb
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-03-29 20:59:04
# Regressor Bug: 1874741
# File Overlap Count: 1
# Description:
#   Bug 1874741 - Part 3: Implement the telemetry to collect GDPR choice on Google Search. r=anti-tracking-reviewers,pbz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D205443
# ==============================================================================

diff -r 1aa93711fe2f -r 7976b0cb00d6 toolkit/components/cookiebanners/components.conf
--- a/toolkit/components/cookiebanners/components.conf	Fri Mar 29 13:03:05 2024 +0000
+++ b/toolkit/components/cookiebanners/components.conf	Fri Mar 29 13:03:06 2024 +0000
@@ -36,4 +36,18 @@
         'constructor': 'CookieBannerListService',
         'processes': ProcessSelector.MAIN_PROCESS_ONLY,
     },
+    {
+        'cid': '{56197e18-d144-45b5-9f77-84102f064462}',
+        'interfaces': ['nsICookieBannerTelemetryService'],
+        'headers': ['/toolkit/components/cookiebanners/nsCookieBannerTelemetryService.h'],
+        'contract_ids': ['@mozilla.org/cookie-banner-telemetry-service;1'],
+        'type': 'mozilla::nsCookieBannerTelemetryService',
+        'singleton': True,
+        'constructor': 'mozilla::nsCookieBannerTelemetryService::GetSingleton',
+        'categories': {
+            'profile-after-change': 'nsCookieBannerTelemetryService',
+            'idle-daily': 'nsCookieBannerTelemetryService',
+        },
+        'processes': ProcessSelector.MAIN_PROCESS_ONLY,
+    },
 ]